src/jsr.rs

```diff
@@ -81,7 +83,15 @@ impl JsrMetadataStore {
       /* checksum */ None,
       |content| {
         let package_info: JsrPackageInfo = serde_json::from_slice(content)?;
-        Ok(Some(Arc::new(package_info)))
+        Ok(Arc::new(package_info))
+      },
+      {
+        let package_name = package_name.to_string();
+        |e| JsrLoadError::PackageManifestLoad(package_name, Arc::new(e))
+      },
+      {
+        let package_name = package_name.to_string();
+        || JsrLoadError::PackageNotFound(package_name)
```

Comment:
This read very strange, I wonder if it would make more sense to assign them to variables before hand. But maybe not very important.